runtime/centrifuge/src/migrations.rs

```diff
@@ -12,4 +12,61 @@
 
 /// The migration set for Centrifuge @ Polkadot.
 /// It includes all the migrations that have to be applied on that chain.
-pub type UpgradeCentrifuge1027 = ();
+pub type UpgradeCentrifuge1028 = migrate_anemoy_external_prices::Migration<super::Runtime>;
+
+mod migrate_anemoy_external_prices {
+	use cfg_primitives::PoolId;
+	use cfg_traits::data::DataRegistry;
+	use cfg_types::oracles::OracleKey;
+	use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+	use pallet_loans::{entities::pricing::ActivePricing, WeightInfo};
+
+	const LOG_PREFIX: &str = "MigrateAnemoyPrices:";
+	const ANEMOY_POOL_ID: PoolId = 4139607887;
+	/// Simply bumps the storage version of a pallet
+	///
+	/// NOTE: Use with caution! Must ensure beforehand that a migration is not
+	/// necessary
+	pub struct Migration<R>(sp_std::marker::PhantomData<R>);
+	impl<R> OnRuntimeUpgrade for Migration<R>
+	where
+		R: pallet_loans::Config<PoolId = PoolId, PriceId = OracleKey>
+			+ pallet_oracle_collection::Config<CollectionId = PoolId, OracleKey = OracleKey>,
+	{
+		fn on_runtime_upgrade() -> Weight {
+			log::info!("{LOG_PREFIX}: STARTING Migrating Anemoy Price Ids.");
+			let active_loans = pallet_loans::ActiveLoans::<R>::get(ANEMOY_POOL_ID);
+			active_loans.clone().into_iter().for_each(|(_, loan)| {
+				if let ActivePricing::External(pricing) = loan.pricing() {
+					match pallet_oracle_collection::Pallet::<R>::register_id(
+						&pricing.price_id(),
+						&ANEMOY_POOL_ID,
+					) {
+						Ok(_) => {
+							log::info!("{LOG_PREFIX}: Registered PriceId: {:?}", pricing.price_id())
+						}
+						Err(e) => log::info!(
+							"{LOG_PREFIX}: Failed to register PriceId: {:?}, with error: {:?}.",
+							pricing.price_id(),
+							e
+						),
```

Comment:
We should only consider it as an error if the loan is `BorrowRestrictions::OraclePriceRequired`. Otherwise, getting an error from `register_id()` is fine. But not sure if for Anemoy we have this case. If we do not see any error right now here, then we do not need to modify it.